backend/src/main/java/com/htecgroup/skynest/exception/UserExceptionHandler.java

```diff
@@ -1,29 +1,55 @@
 package com.htecgroup.skynest.exception;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.htecgroup.skynest.model.response.ErrorMessage;
 import com.htecgroup.skynest.util.DateTimeUtil;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.validation.FieldError;
 import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.context.request.WebRequest;
 
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
 import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
 
 @ControllerAdvice
 @Slf4j
 public class UserExceptionHandler {
 
+  public static ResponseEntity<ErrorMessage> responseFor(UserException ex) {
```

Comment:
If you use this methods in some other classes, remove then from here.. `ContollerAdvice` has just one purpose and that is to handle all unhandled or purposely thrown exceptions and return generic response.

On the other hand if you want some method from ControllerAdvice to be executed, then throw that custom exception from that class that you handle

Reply:
make it a component if that is the problem

Reply:
I've tried adding Component or Service annotations to the filter, but it doesn't work.

Reply:
You can extract that in separate class then, but not like this now.. Maybe it will be good if creating od ErrorMessage is in util class in a method and you just reference to the same logic from both places